app/services/date_stamper.rb

```diff
@@ -22,8 +22,9 @@ def call
     # -- If case type is “date stampable” then we use the date stamp value
     # -- If case is non “date stampable” we use the submission date as the date
     #    stamp.
+    return false if @crime_app.date_stamp.present?
 
-    if CaseType.new(@case_type).date_stampable? && @crime_app.date_stamp.nil?
+    if @crime_app.not_means_tested? || CaseType.new(@case_type).date_stampable?
```

Comment:
How comes we don't need the nil check as well?

Reply:
Bc the nil check is covered on line 25

I wanted to save the cognitive overhead of brackets

Reply:
It was:
If it's a case type that should get a datestamp, AND doesn't already have a date stamp, give it a datestamp

Now:
If it already has a datestamp return false
THEN
If it's a non means tested app OR  a case type that should get a datestamp, give it a datestamp